endpoints/openrtb2/video_auction.go

```diff
@@ -147,13 +147,26 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 	//create full open rtb req from full video request
 	mergeData(videoBidReq, bidReq)
 
+	initialPodNumber := len(videoBidReq.PodConfig.Pods)
+	if len(podErrors) > 0 {
+		//remove incorrect pods
+		videoBidReq = cleanupVideoBidRequest(videoBidReq, podErrors)
+	}
+
 	//create impressions array
-	imps, errs := deps.createImpressions(videoBidReq)
-	if errs != nil {
-		errL = append(errL, errs...)
+	imps, podErrors := deps.createImpressions(videoBidReq, podErrors)
+
+	if len(podErrors) == initialPodNumber {
```

Comment:
Mansi, thank you for your comments! 
Good catch, `len(imps) == 0` will do the same, but logically it will more tangled, because one pod may require more than one impression to be created. `podErrors` is a list of errors per pod and this code handles the case when all pods are incorrect. Also pod error may occur not just because of impression creation error. I would not mix this logic. 
`initialPodNumber` variable is initialized before any incorrect pods clean up, so I don't see any problems with this.  
Please let me know what you think.

Reply:
Ah! I see what you mean about podErrors. Initially, I missed that podErrors was being passed to `createImpressions` and was thinking that `podErrors` returned from `createExpressions` might not reflect errors for all pods but only for the pods left after the clean up on line 153. You are right, this should be fine.

Also, regarding your comment about one pod potentially requiring  more than one imps, if all pods were incorrect then `createImpressions` should return `0` imps, right? And so `len(imps) == 0` should also give the same result, correct? (You don't need to change it to `len(imps)==0`. I am just making sure I understood you correctly)

Reply:
I've just tried it and it all works including unit tests! Which is a good sign, we don't need to process request if we have no impressions to send downstream. 
Thank you for this, good catch!